optuna/visualization/_optimization_history.py

```diff
@@ -87,36 +102,169 @@ def _get_optimization_history_plot(
         yaxis={"title": target_name},
     )
 
-    trials = [t for t in study.trials if t.state == TrialState.COMPLETE]
+    if len(studies) == 0:
+        _logger.warning("There are no studies.")
+        return go.Figure(data=[], layout=layout)
+
+    all_trials = list(
+        itertools.chain.from_iterable(
+            (
+                trial
+                for trial in study.get_trials(deepcopy=False)
+                if trial.state == TrialState.COMPLETE
+            )
+            for study in studies
+        )
+    )
 
-    if len(trials) == 0:
-        _logger.warning("Study instance does not contain trials.")
+    if len(all_trials) == 0:
+        _logger.warning("There are no complete trials.")
         return go.Figure(data=[], layout=layout)
 
-    if target is None:
-        if study.direction == StudyDirection.MINIMIZE:
-            best_values = np.minimum.accumulate([cast(float, t.value) for t in trials])
-        else:
-            best_values = np.maximum.accumulate([cast(float, t.value) for t in trials])
-        traces = [
-            go.Scatter(
-                x=[t.number for t in trials],
-                y=[t.value for t in trials],
-                mode="markers",
-                name=target_name,
-            ),
-            go.Scatter(x=[t.number for t in trials], y=best_values, name="Best Value"),
+    if error_bar:
+        return _get_optimization_history_with_error_bar(studies, target, target_name, layout)
```

Comment:
[question]
Why is `optimization_history` in the method name a singular form?

Reply:
For consistency to `_get_optimization_histories`, how about renaming this?

```suggestion
        return _get_optimization_histories_with_error_bar(studies, target, target_name, layout)
```